pyERGM/metrics.py

```diff
@@ -461,6 +574,39 @@ def calculate_for_sample(networks_sample: np.ndarray | torch.Tensor):
             raise ValueError(f"Unsupported type of sample: {type(networks_sample)}! Supported types are np.ndarray and "
                              f"torch.Tensor with is_sparse=True")
 
+    def calculate_bootstrapped_features(self, first_halves_to_use: np.ndarray,
+                                        second_halves_to_use: np.ndarray,
+                                        first_halves_indices: np.ndarray[int], second_halves_indices: np.ndarray[int]):
+        """
+        Calculates the bootstrapped number of reciprocal dyads, by counting such pairs in the sampled subnetworks, and
+        normalizing by network size (i.e., calculating the fraction of existing reciprocal dyads out of all possible
+        ones in sampled subnetworks, and multiplying by the number of possible reciprocal dyads in the full observed
+        network).
+        Parameters
+        ----------
+        first_halves_to_use
+            Multiple samples of subnetworks of an observed network, representing the connectivity between half of the
+            nodes in the large network.
+        second_halves_to_use
+            The subnetworks formed by the complementary set of nodes of the large network for each sample.
+        first_halves_indices
+            The indices of the nodes in the first half of the large network for each sample, according to the ordering
+            of the nodes in the large network.
+        second_halves_indices
+            The indices of the nodes in the second half of the large network for each sample, according to the ordering
+
+        Returns
+        -------
+        Properly normalized statistics of subnetworks of an observed network.
+        """
+        # TODO: I think it's identical to the function of NumberOfEdgesUndirected, consider replicating code.
+        num_nodes_in_observed = first_halves_indices.shape[0] + second_halves_indices.shape[0]
+        num_nodes_in_first_half = first_halves_indices.shape[0]
+        num_possible_rec_dyads_observed = num_nodes_in_observed * (num_nodes_in_observed - 1) / 2
+        num_possible_rec_dyads_first_half = num_nodes_in_first_half * (num_nodes_in_first_half - 1) / 2
+        return self.calculate_for_sample(
```

Comment:
Same question as in NumEdges - why not take the mean with the calculation of the second half?

Reply:
Same answer, let's decide what to do and do that in both cases.